zerver/views/streams.py

```diff
@@ -386,19 +386,18 @@ def update_stream_backend(
 
     for setting_name, permissions_configuration in Stream.stream_permission_group_settings.items():
         request_settings_dict = locals()
-        setting_group_id_name = setting_name + "_id"
 
-        if setting_group_id_name not in request_settings_dict:  # nocoverage
+        if setting_name not in request_settings_dict:  # nocoverage
             continue
 
-        if request_settings_dict[setting_group_id_name] is not None and request_settings_dict[
-            setting_group_id_name
+        if request_settings_dict[setting_name] is not None and request_settings_dict[
+            setting_name
```

Comment:
Hmm, this logic looks wrong to me -- won't `getattr(stream, setting_name)` always be a `Stream` object, not an ID?

I think we would do better preserving the previous model of having clear variable names like `setting_group_id_name` and using the `REQ` support for having the API name be different from the function parameter name.

Reply:
Yes, `getattr(stream, "can_remove_subscribers_group")` should return a group object while `getattr(stream, "can_remove_subscribers_group_id")` should return group ID.

Reply:
Though, it is not clear to me what you mean by using `REQ` support. Do you mean doing something like we do in https://github.com/zulip/zulip/blob/befcb46ba39a4e37d6ed1b73b362a2d657d1e4c5/zerver/views/realm.py#L74?

If yes, then I guess I did something similar for one of the existing group settings and you suggested to avoid doing this. I am not sure though as I cannot find the PR where I think you suggested so.

Reply:
Yes, this is what I'd had in mind. I think the reason it'd be nice here is for typing -- we can have the `_id` name be for the integer, and the bare name for a `UserGroup` object.